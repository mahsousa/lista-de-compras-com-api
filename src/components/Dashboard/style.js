import styled from "styled-components";


export const Container = styled.div`
    // max-width:1120px;
    gap:20px;
    // justify-content: space-around;

    @media (max-width:320px){
        display: flex;
        flex-wrap: wrap;
        gap:0;
        justify-content: center;
        margin: 10px 0;
      }

      @media (min-width: 481px) and (max-width: 991px){
        display:flex;
        flex-wrap:wrap;
        gap:0;
        margin: 20px 0;
      }
`;
