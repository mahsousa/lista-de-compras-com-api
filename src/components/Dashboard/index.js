import React from 'react'
import DashItem from '../DashItem';
import Assets from '../Assets';
import * as C from "./style"; 
import{
  FaRegArrowAltCircleUp,
  FaRegArrowAltCircleDown,
  FaDollarSign,
} from "react-icons/fa";



const Dashboard = ({entrada, saida, total}) => {
  const totalNum =  parseFloat(total);
  return (
      <C.Container>
        <DashItem title="Total OrÃ§amento"  Icon={FaRegArrowAltCircleUp} iconColor="green" value={entrada}/>
        <DashItem title="Total Gasto" Icon={FaRegArrowAltCircleDown} iconColor="red" value={saida}/>
        <DashItem title="Disponivel" Icon={FaDollarSign} value={total} textoRed={ totalNum < 0 ? true : false}/>
        <Assets key="assets"/>
      </C.Container>
  );
};
export default Dashboard;