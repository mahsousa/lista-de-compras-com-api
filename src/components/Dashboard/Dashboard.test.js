import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';


import Dashboard from '.';

jest.mock("@react-pdf/renderer", () => {
  const React = require("react");

  const PDFDownloadLink = ({ document, fileName, children }) => {
    const mockOnClick = () => {
      // Fazer algo quando o link for clicado
    };

    return (
      <div onClick={mockOnClick}>
        {children}
      </div>
    );
  };

  const AiFillPrinter = () => {
    return <div>Mocked AiFillPrinter</div>;
  };

  return {
    __esModule: true,
    Font: {
      register: jest.fn(),
    },
    Image: jest.fn(),
    Page: jest.fn(({ children }) => <div>{children}</div>),
    Text: jest.fn(),
    View: jest.fn(),
    Document: jest.fn(({ children }) => <div>{children}</div>),
    StyleSheet: {
      create: jest.fn((styles) => styles),
    },
    PDFDownloadLink: PDFDownloadLink,
    AiFillPrinter: AiFillPrinter,
  };
});


it('renders the component value correctly', () => {
  const component = renderer.create(<Dashboard entrada="10" saida="10" total="10" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


