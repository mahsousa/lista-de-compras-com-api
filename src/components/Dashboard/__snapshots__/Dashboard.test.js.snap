// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component value correctly 1`] = `
<div
  className="sc-gLDzan fKDaMW"
>
  <div
    className="sc-beqWaB dYSuyS"
  >
    <header
      className="sc-gueYoa jJHxPM"
    >
      <p
        className="sc-dmqHEX dOYHvZ"
      >
        Total Or√ßamento
      </p>
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "green",
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 504c137 0 248-111 248-248S393 8 256 8 8 119 8 256s111 248 248 248zm0-448c110.5 0 200 89.5 200 200s-89.5 200-200 200S56 366.5 56 256 145.5 56 256 56zm20 328h-40c-6.6 0-12-5.4-12-12V256h-67c-10.7 0-16-12.9-8.5-20.5l99-99c4.7-4.7 12.3-4.7 17 0l99 99c7.6 7.6 2.2 20.5-8.5 20.5h-67v116c0 6.6-5.4 12-12 12z"
        />
      </svg>
    </header>
    <span
      className="sc-hLseeU ckMyXo"
    >
       R$ 
      10
    </span>
  </div>
  <div
    className="sc-beqWaB dYSuyS"
  >
    <header
      className="sc-gueYoa jJHxPM"
    >
      <p
        className="sc-dmqHEX dOYHvZ"
      >
        Total Gasto
      </p>
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "red",
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm-32-316v116h-67c-10.7 0-16 12.9-8.5 20.5l99 99c4.7 4.7 12.3 4.7 17 0l99-99c7.6-7.6 2.2-20.5-8.5-20.5h-67V140c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12z"
        />
      </svg>
    </header>
    <span
      className="sc-hLseeU ckMyXo"
    >
       R$ 
      10
    </span>
  </div>
  <div
    className="sc-beqWaB dYSuyS"
  >
    <header
      className="sc-gueYoa jJHxPM"
    >
      <p
        className="sc-dmqHEX dOYHvZ"
      >
        Disponivel
      </p>
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 288 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M209.2 233.4l-108-31.6C88.7 198.2 80 186.5 80 173.5c0-16.3 13.2-29.5 29.5-29.5h66.3c12.2 0 24.2 3.7 34.2 10.5 6.1 4.1 14.3 3.1 19.5-2l34.8-34c7.1-6.9 6.1-18.4-1.8-24.5C238 74.8 207.4 64.1 176 64V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48h-2.5C45.8 64-5.4 118.7.5 183.6c4.2 46.1 39.4 83.6 83.8 96.6l102.5 30c12.5 3.7 21.2 15.3 21.2 28.3 0 16.3-13.2 29.5-29.5 29.5h-66.3C100 368 88 364.3 78 357.5c-6.1-4.1-14.3-3.1-19.5 2l-34.8 34c-7.1 6.9-6.1 18.4 1.8 24.5 24.5 19.2 55.1 29.9 86.5 30v48c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-48.2c46.6-.9 90.3-28.6 105.7-72.7 21.5-61.6-14.6-124.8-72.5-141.7z"
        />
      </svg>
    </header>
    <span
      className="sc-hLseeU ckMyXo"
    >
       R$ 
      10
    </span>
  </div>
  <div
    className="sc-jTrPJq ggCkkf"
  >
    <div
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "#215581",
            "height": "40px",
            "padding": "5px",
            "width": "40px",
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M732 120c0-4.4-3.6-8-8-8H300c-4.4 0-8 3.6-8 8v148h440V120zm120 212H172c-44.2 0-80 35.8-80 80v328c0 17.7 14.3 32 32 32h168v132c0 4.4 3.6 8 8 8h424c4.4 0 8-3.6 8-8V772h168c17.7 0 32-14.3 32-32V412c0-44.2-35.8-80-80-80zM664 844H360V568h304v276zm164-360c0 4.4-3.6 8-8 8h-40c-4.4 0-8-3.6-8-8v-40c0-4.4 3.6-8 8-8h40c4.4 0 8 3.6 8 8v40z"
        />
      </svg>
    </div>
    <svg
      fill="currentColor"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": "#215581",
          "height": "40px",
          "padding": "5px",
          "width": "30px",
        }
      }
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
      />
    </svg>
  </div>
</div>
`;
